### PR TITLE
tables: make responsive styling apply to all stackable tables

### PR DESCRIPTION
**Pull request by jennur**
Closes https://github.com/inveniosoftware/react-invenio-deposit/issues/531

Needs https://github.com/inveniosoftware/react-invenio-deposit/pull/536 and
https://github.com/inveniosoftware/invenio-communities/pull/736

- Removed custom responsive classes
- Applied the same mobile/tablet styling to all tables so that the layout stays consistent and works well out of the box when/if adding new tables
- Moved some css classes from `table.variables` to `table.overrides`

### Deposit form
**Desktop** (not changed)
Screenshot 2022-06-29 at 13 56 53

**Tablet**
Screenshot 2022-06-29 at 13 57 08

**Mobile**
Screenshot 2022-06-29 at 13 57 19

**Edit of published record (Tablet)**
Screenshot 2022-06-29 at 14 05 46

### Landing page
**Desktop** (Not changed)
Screenshot 2022-06-29 at 13 57 34

**Tablet**
Screenshot 2022-06-29 at 13 57 42

**Mobile**
Screenshot 2022-06-29 at 13 57 50

### Members

**Desktop** (Not changed)
Screenshot 2022-06-29 at 14 31 52

**Tablet**
Screenshot 2022-06-29 at 14 32 04

**Mobile**
Screenshot 2022-06-29 at 14 32 34

### Invitations
**Desktop** (Not changed)
Screenshot 2022-06-29 at 13 56 13

**Tablet**
Screenshot 2022-06-29 at 13 56 21

**Mobile**
Screenshot 2022-06-29 at 13 56 30

### CODE REVIEW

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/table.overrides [hunk 1]
```diff
@@ -3,68 +3,94 @@
...
+              background-color: lighten(@borderColor, 75%);
+            }
+
+            &:not(.right.aligned) {
+              text-align: left;
+            }
+
+            &[data-label]  {
+              display: inline-grid !important;
+              grid-template-columns: [col1-start] 30% [col2-start] 70% [end];
```
Comment (Pineirin) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/table.overrides:67]: Why did you need to define the columns here instead of using the Grid from semantic-ui?

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/views/item.overrides [hunk 2]
```diff
@@ -112,7 +112,7 @@
 .item {
 
   &.align-no-checkbox {
-    padding-left: @defaultPadding / 2 + @checkboxSize;
+    padding-left: @defaultMargin / 2 + @checkboxSize;
```
[the remark this answers is not in the document]
Comment (jennur) [reply invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/views/item.overrides:115]: It's just so that it stays aligned with the version with the check-box (that one has a checkbox element and `@defaultMargin / 2`)

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/table.overrides (hunk 1 above)
Comment (jennur) [reply invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/table.overrides:67]: Because the `td[data-label]` only has a `::before`, it's not an actual element, so it's not controllable like that. Also, this styling is to make everything work for `.ui.table`s out of the box so that there is no need to customize the markup to make it responsive (other than adding a `data-label`-attribute to `td`s that should have a label for the mobile version :)

**Review by kpsherva — APPROVED** [review]
